source/client-side-encryption/tests/unified/namedKMS-rewrapManyDataKey.yml

```diff
@@ -0,0 +1,432 @@
+              databaseName: *database0Name
+              command:
+                update: *collection0Name
+                ordered: true
+                updates:
+                  - q: { _id: { $$type: binData } }
+                    u: { $set: { masterKey: { provider: "local:name1" }, keyMaterial: { $$type: binData } }, $currentDate: { updateDate: true } }
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+                  - q: { _id: { $$type: binData } }
+                    u: { $set: { masterKey: { provider: "local:name1" }, keyMaterial: { $$type: binData } }, $currentDate: { updateDate: true } }
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+                  - q: { _id: { $$type: binData } }
+                    u: { $set: { masterKey: { provider: "local:name1" }, keyMaterial: { $$type: binData } }, $currentDate: { updateDate: true } }
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+                  - q: { _id: { $$type: binData } }
+                    u: { $set: { masterKey: { provider: "local:name1" }, keyMaterial: { $$type: binData } }, $currentDate: { updateDate: true } }
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+                writeConcern: { w: majority }
+
+  - description: "rewrap from local:name1 to local:name2"
+    operations:
+      - name: rewrapManyDataKey
+        object: *clientEncryption0
+        arguments:
+          filter: { keyAltNames: { $eq: "local:name1_key" } }
+          opts:
+            provider: "local:name2"
+        expectResult:
+          bulkWriteResult:
+            insertedCount: 0
+            matchedCount: 1
+            modifiedCount: 1
+            deletedCount: 0
+            upsertedCount: 0
+            upsertedIds: {}
+            insertedIds: { $$unsetOrMatches: {} }
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              databaseName: *database0Name
+              command:
+                find: *collection0Name
+                filter: { keyAltNames: { $eq: "local:name1_key" } }
+                readConcern: { level: majority }
+          - commandStartedEvent:
+              databaseName: *database0Name
+              command:
+                update: *collection0Name
+                ordered: true
+                updates:
+                  - q: { _id: { $$type: binData } }
+                    u: { $set: { masterKey: { provider: "local:name2" }, keyMaterial: { $$type: binData } }, $currentDate: { updateDate: true } }
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+                writeConcern: { w: majority }
+
+  - description: "rewrap from aws:name1 to aws:name2"
```

Comment:
This test fails in python with:
```
 [2024/01/26 09:02:28.906] test/unified_format.py:1280: in _clientEncryptionOperation_rewrapManyDataKey
 [2024/01/26 09:02:28.906]     data = target.rewrap_many_data_key(*args, **kwargs)
 [2024/01/26 09:02:28.906] pymongo/encryption.py:1057: in rewrap_many_data_key
 [2024/01/26 09:02:28.906]     with _wrap_encryption_errors():
 [2024/01/26 09:02:28.906] /opt/python/3.10/lib/python3.10/contextlib.py:153: in __exit__
 [2024/01/26 09:02:28.906]     self.gen.throw(typ, value, traceback)
 [2024/01/26 09:02:28.906] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 [2024/01/26 09:02:28.906]     @contextlib.contextmanager
 [2024/01/26 09:02:28.906]     def _wrap_encryption_errors() -> Iterator[None]:
 [2024/01/26 09:02:28.906]         """Context manager to wrap encryption related errors."""
 [2024/01/26 09:02:28.906]         try:
 [2024/01/26 09:02:28.906]             yield
 [2024/01/26 09:02:28.906]         except BSONError:
 [2024/01/26 09:02:28.906]             # BSON encoding/decoding errors are unrelated to encryption so
 [2024/01/26 09:02:28.906]             # we should propagate them unchanged.
 [2024/01/26 09:02:28.906]             raise
 [2024/01/26 09:02:28.906]         except Exception as exc:
 [2024/01/26 09:02:28.906] >           raise EncryptionError(exc) from None
 [2024/01/26 09:02:28.906] E           pymongo.errors.EncryptionError: Error in KMS response. HTTP status=400. Response body=
 [2024/01/26 09:02:28.906] E           {"__type":"AccessDeniedException","Message":"User: arn:aws:iam::579766882180:user/BUILD-7242 is not authorized to perform: kms:Encrypt on this resource because the resource does not exist in this Region, no resource-based policies allow access, or a resource-based policy explicitly denies access"}
```
https://spruce.mongodb.com/task/mongo_python_driver_tests_python_version_rhel8_test_encryption__platform~rhel8_auth_ssl~noauth_nossl_python_version~3.10_encryption~encryption_crypt_shared_test_5.0_replica_set_patch_41a131ea1c15ffa969a14ce1334ce19837dc226b_65b3e49a57e85a1e221c98f3_24_01_26_16_58_03?execution=0&sortBy=STATUS&sortDir=ASC

Do you know what the issue could be?

Reply:
I expect this is due to `aws:name2` being configured with the credentials from `aws:name1`. The `aws:name1` account deliberately cannot access the key in the `aws:name2` account to ensure rewrapping uses separate accounts.

See the PR description for steps to get the `aws:name2` credentials:

> Test credentials for `aws:name2` are available in AWS Secrets Manager under `drivers/client-side-encryption`. The `aws:name2` account credentials start with `AWSNAME2_`. See https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets for more background on how the secrets are managed.

Reply:
Thanks this makes sense now.